extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -237,14 +271,24 @@ export class ResourceTypeService implements IResourceTypeService {
 	private getOkButtonText(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): string | undefined {
 		if (resourceType.okButtonText) {
 			for (const possibleOption of resourceType.okButtonText) {
-				if (processWhenClause(possibleOption.when, selectedOptions)) {
+				if (possibleOption && processWhenClause(possibleOption.when, selectedOptions)) {
```

Comment:
When do you expect possibleOption to be undefined here?

Reply:
For example, the miaa sub resource type does not define an okaybuttontext for itself. In those situations, I am appending an undefined okButtonText to the okButtonArray.

Reply:
Why wouldn't you just have the array be empty? That'd make more sense to me.

Reply:
So for example there are 3 subresources (options) for a resource. If second has described some okbuttonText, the array will look like

[undefined, okButtonText, undefined]

Reply:
Nevermind, we are always using the 'when' clause this is not needed.